app/assets/javascripts/map.js

```diff
@@ -0,0 +1,210 @@
+//= require polyline
+var map;
+
+var directionsService = new google.maps.DirectionsService();
+
+function initialize() {
+  directionsDisplay = new google.maps.DirectionsRenderer();
+
+  var mapOptions = {
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.ROADMAP
+  };
+
+  map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
+  directionsDisplay.setMap(map);
+  route = new Route();
+  route.calculateRoute();
+
+}
+
+//-------Route----------
+
+var Route = function(){
+  this.start = new google.maps.LatLng(41.953819, -87.654750), // Chicago for now
+  this.end = new google.maps.LatLng(38.637548, -90.205010), // St. Louis for now
+  this.waypts = this.getWaypoints();
+};
+
+Route.prototype.getWaypoints = function(){
+  var waypts = [];
+  var checkboxArray = document.getElementById('waypoints');
+
+  for (var i = 0; i < checkboxArray.length; i++) {
+    if (checkboxArray.options[i].selected == true) {
+      waypts.push({
+        location: checkboxArray[i].value,
+        stopover: true});
+    }
+  }
+
+  return waypts;
+};
+
+Route.prototype.calculateRoute = function(){
+  var request = {
+    origin: this.start,
+    destination: this.end,
+    waypoints: this.waypts,
+    optimizeWaypoints: true,
+    travelMode: google.maps.TravelMode.DRIVING
+  };
+
+  directionsService.route(request, function(response, status){
+    if (status == google.maps.DirectionsStatus.OK) {
+
+      decoder = new PolylineDecoder()
+      route.polyline = decoder.decodePolyline(response);
+      polygon = new Polygon(route.polyline)
+      route.displayRoute(response, true)
+    }
+  });
+};
+
+
+Route.prototype.displayRoute = function(response, drawMe){
+  drawMe = typeof drawMe !== 'undefined' ? drawMe : true;
+
+  directionsDisplay.setDirections(response);
+
+  if (drawMe) {
+    drawer = new Drawer();
+    drawer.draw(polygon.geoJson);
+  };
+}
+
+
+//----polyline decoder-----
```

Comment:
Route and Polyline should really be in their own files.  With good encapsulation of objects this should totally be possible -- you'll have an `App` or other such object that acts as the controller and sets up the relationship between the objects during `$(document).ready`